src/Instances/Attribute.lua

```diff
@@ -0,0 +1,49 @@
+--!strict
+
+--[[
+	A special key for property tables, which allows users to apply custom
+    attributes to instances
+]]
+
+local Package = script.Parent.Parent
+local PubTypes = require(Package.PubTypes)
+local logError = require(Package.Logging.logError)
+local xtypeof = require(Package.Utility.xtypeof)
+local Observer = require(Package.State.Observer)
+
+local function setAttribute(instance: Instance, attribute: string, value: any)
+    instance:SetAttribute(attribute, value)
+end
+
+local function bindAttribute(instance: Instance, attribute: string, value: any, cleanupTasks: {PubTypes.Task})
+    if xtypeof(value) == "State" then
+		setAttribute(instance, attribute, value:get(false))
+		table.insert(cleanupTasks, Observer(value :: any):onChange(function()
```

Comment:
I'm not super familiar with Luau typing but does `value` need to be casted to `any` when its type (line 18) already is set to any?

Reply:
Not sure on this one. I read on some of the other functions and they seem to cast it to any as well. If it's not conventional then I might change it back